snyk/managers.py

```diff
@@ -292,7 +289,7 @@ def all(self) -> Any:
         )
         resp = self.client.get(path)
         dependency_data = resp.json()
-        return self.klass.from_dict(dependency_data)
+        return self.klass.from_dict(dependency_data["depGraph"])
```

Comment:
This looks correct, except it's missing a check for the presence of that key, and will throw a `KeyError` exception if it's not in the response. Would be good to add a simple gate.

Here's an existing example of the same pattern https://github.com/snyk-samples/pysnyk/blob/master/snyk/managers.py#L258-L261

Note information in exceptions needs to be improved, but we can do that all in one go.

Reply:
I don't think this gate is needed. According to the schema `depGraph` is required and should always be there.
It is different for the example you shared. `jiraIssue` is not marked as `required` in the schema and could indeed be not present.

Reply:
That assumes everything is working fine. In the case that it's not, or something about the API changes, it would be better to throw an exception we control than just a generic error.